src/Trakx.CoinGecko.ApiClient/CoinGeckoClient.cs

```diff
@@ -47,183 +38,226 @@ public class CoinGeckoClient : ICoinGeckoClient
-            quoteResponse.Content.Market_data.Current_price[Constants.Usd] : default;
+    /// <summary>
+    /// As of 2023-06-23, CoinGecko does not support USDc as a quote currency.
+    /// <see href="https://api.coingecko.com/api/v3/simple/supported_vs_currencies"/>
+    /// As such, we need to:
+    /// <list type="bullet">
+    /// <item><description>Use USD as the 'vs currency' in the API call</description></item>
+    /// <item><description>Also get the price of the wanted quote currency</description></item>
+    /// <item><description>Convert the prices to the wanted quote currency</description></item>
+    /// </list>
+    /// </summary>
+    public async Task<decimal?> GetLatestPrice(
+        string coinGeckoId,
+        string quoteCurrencyId = Constants.UsdCoin,
+        CancellationToken cancellationToken = default)
+    {
+        Guard.Against.NullOrWhiteSpace(coinGeckoId);
+        Guard.Against.NullOrWhiteSpace(quoteCurrencyId);
 
-        if (fxRate == null)
-        {
-            Logger.LogDebug($"Current price for '{Constants.Usd}' in coin id '{quoteCurrencyId} for date '{date:dd-MM-yyyy}' is missing.");
-            throw new FailedToRetrievePriceException($"Failed to retrieve price of {quoteCurrencyId} as of {date}");
-        }
+        var prices = await GetAllPrices(
+            coinGeckoId.AsSingletonIEnumerable(),
+            quoteCurrencyId.AsSingletonArray(),
+            cancellationToken);
 
-        var entry = _cache.CreateEntry(cacheKey);
-        entry.AbsoluteExpirationRelativeToNow = TimeSpan.FromDays(1);
-        entry.Value = fxRate.Value;
-        return fxRate.Value;
+        var price = prices.GetPrice(coinGeckoId, quoteCurrencyId);
+        return price;
     }
 
     /// <inheritdoc />
-    public async Task<MarketData?> GetMarketDataAsOfFromId(string id, DateTime asOf, string quoteCurrencyId = "usd-coin")
+    public async Task<MultiplePrices> GetAllPrices(
+        IEnumerable<string> ids,
+        string[]? vsCurrencies = default,
+        CancellationToken cancellationToken = default)
     {
-        var date = asOf.ToString("dd-MM-yyyy");
-        var cacheKey = $"{_typeName}|market-data|{id}|{quoteCurrencyId}|{date}";
-        if (_cache.TryGetValue(cacheKey, out MarketData cached)) return cached;
+        Guard.Against.Null(ids);
 
-        var fullData = await _coinsClient
-            .HistoryAsync(id, date, false.ToString())
-            .ConfigureAwait(false);
+        var supportedCurrencies = await GetSupportedQuoteCurrencies(cancellationToken);
 
-        var fxRate = await GetUsdFxRate(quoteCurrencyId, date);
+        (var baseIds, var quoteIds) = GetIdsForPriceQuery(ids, vsCurrencies, supportedCurrencies);
 
-        if (fullData.Content.Market_data == null)
-            return null;
+        var response = await _simpleClient
+            .PriceAsync(baseIds, quoteIds, cancellationToken: cancellationToken)
```

Comment:
Here did you test with a fairly big list of base currencies? I think I remember seeing it failing with an error like 
`input string too long, should be less that 1024 characters` or something like that

Reply:
https://trakx.atlassian.net/browse/TKX-860

I ran a test directly against the API, via `Postman`.
It worked OK for 1099 chars.

Repeating my comment from JIRA:
The current PR will be submitted without a “pagination” of symbols when getting prices, I will add one after if that’s OK.

So if you agree, my plan is:
- have the current (already meaty) PR merged and released ASAP
- return to the `market-data` PR, to get `CoinGecko` in bulk
- once the `CoinGecko` usage is tamed, I return to `coingecko-api-client` and add support for a limitless list of tokens.

Reply:
Test added in #104